File: Easy/Reverse Words In A String Iii/ReverseWordsInAStringIii.js
// https://leetcode.com/problems/reverse-words-in-a-string-iii

/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
  return s.split(" ").map(reverseString).join(" ");
};

function reverseString(s) {
  return s.split("").reverse().join("");
}

File: Easy/Reverse Words In A String Iii/ReverseWordsInAStringIii.cpp
// https://leetcode.com/problems/reverse-words-in-a-string-iii

class Solution {
public:
  string reverseWords(string s) {
    int pos = s.find(' ', 0);
    int prevpos = 0;
    while (pos != string::npos) {
      reverse(s.begin() + prevpos, s.begin() + pos);
      prevpos = pos + 1;
      pos = s.find(' ', prevpos);
    }
    reverse(s.begin() + prevpos, s.end());
    return s;
  }
};